frontend/console/src/features/modules/schema/schema.utils.ts

```diff
@@ -28,3 +71,56 @@ export function isFirstLineOfBlock(ll: string[], i: number): boolean {
   }
   return staticKeywords.includes(tokens[0]) || declTypes.includes(tokens[0])
 }
+
+export interface DeclSchema {
+  schema: string
+  declType: string
+}
+
+export function declFromModules(moduleName: string, declName: string, modules: UseQueryResult<GetModulesResponse, Error>) {
+  if (!modules.isSuccess || modules.data.modules.length === 0) {
+    return
+  }
+  const module = modules.data.modules.find((module) => module.name === moduleName)
+  if (!module?.schema) {
+    return
+  }
+  return declFromModuleSchemaString(declName, module.schema)
+}
+
+export function declFromModuleSchemaString(declName: string, schema: string) {
+  const lines = schema.split('\n')
+  const foundIdx = lines.findIndex((line) => {
+    const regex = new RegExp(`^  (export )?\\w+ ${declName}`)
+    return line.match(regex)
+  })
+
+  if (foundIdx === -1) {
+    return
+  }
+
+  const line = lines[foundIdx]
+  let out = line
+  let subLineIdx = foundIdx + 1
+  while (subLineIdx < lines.length && lines[subLineIdx].startsWith('    ')) {
```

Comment:
nit: does it make sense to do `startsWith(' (\\w+)')` here as below instead of 4 spaces?

Reply:
Great question! So the behavior here is slightly different - basically, the `(\\w+)` extracts the decl type from the line but ignores the start of the line, whereas this line is just checking the nesting depth from the start of the line.